### PR TITLE
This commit fixes IPv4 and adds IPv6 pinhole to nftables

### PR DESCRIPTION
**Pull request by svenauhagen**
Hi,

this commit makes nftables usable and adds IPv6 pinhole support.

Best
Sven

### CODE REVIEW

File: miniupnpd/netfilter_nft/nftnlrdr.c [hunk 1]
```diff
@@ -87,7 +89,7 @@ remove_timestamp_entry(unsigned short eport, int proto)
 	p = &timestamp_list;
 	e = *p;
 	while(e) {
-		if(e->eport == eport && e->protocol == (short)proto) {
+		if (e->eport == eport && e->protocol == (short)proto) {
```
Comment (miniupnp) [inline miniupnpd/netfilter_nft/nftnlrdr.c]: I don 't like much when a commit mixes functional and aesthetic changes

File: miniupnpd/netfilter_nft/nftnlrdr.c [hunk 2]
```diff
@@ -177,13 +183,16 @@ add_filter_rule2(const char * ifname,
...
-			    desc, 0);
-	return nft_send_request(r, NFT_MSG_NEWRULE);
+        rhost_addr = inet_addr(rhost);
+    }
+
+	r = rule_set_filter(NFPROTO_INET, ifname, proto,
+			    rhost_addr, inet_addr(iaddr),
+				eport, iport, 0, desc, 0);
+
+	return nft_send_request(r, NFT_MSG_NEWRULE, RULE_CHAIN_FILTER);
```
Comment (miniupnp) [inline miniupnpd/netfilter_nft/nftnlrdr.c]: it is difficult to understand what is changed (many changes for just indentation, etc.)

**Comment by miniupnp** [comment]
@svenauhagen would it be possible to check build with travis ?

**Comment by miniupnp** [comment]
with which linux version has this been tested ?

**Comment by svenauhagen** [comment]
@miniupnp 
I can also try and break up the commit in smaller chunks if it makes the review easier.
I do not have a travis active on my account, is there one on the miniupnp branch?
I am using Jenkins internally.

I have tested and compiled it with Debian Stretch and a Backports Kernel and nftables.
nftables needs to be fairly new for this to work.
It should work out of the box on a newer system like Debian Buster or a recent Ubuntu Version.

**Comment by miniupnp** [comment]
you can easily activate travis-ci for your github repositories.

if a recent version of nftables is needed, I would prefer some #if to allow building on old or new nftables.
There is the same thing with PF or iptables in miniupnpd

aslo please rebase on my master :)

**Comment by svenauhagen** [comment]
Hi,

alright, I rebased the PR and fixed some indentation problems.
I enabled travis on my account and just started the build.

Regarding the version that was just so that you need a Kernel that actually supports enough of nftables to run the code.
There is no difference inside the committed code regarding the different nftable versions.

**Comment by miniupnp** [comment]
what is that problem ? 
```
netfilter_nft/nftnlrdr.c: In function ‘delete_filter_rule’:
netfilter_nft/nftnlrdr.c:221:2: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
  LIST_FOREACH(p, &head_filter, entry) {
  ^
```

**Comment by miniupnp** [comment]
by the way, I don't see any rebase, just a merge...

**Comment by miniupnp** [comment]
I rebased myself...

**Comment by svenauhagen** [comment]
sorry about that I did a merge not a rebase.
I found the cause of the compiler warnings.
It was the typedef struct definition.
The warnings were in the code from the beginning so that was nothing I introduced.

**Comment by miniupnp** [comment]
Le 13/06/2019 à 07:30, sven a écrit :
>
> sorry about that I did a merge not a rebase.
> I found the cause of the compiler warnings.
> It was the typedef struct definition.
> The warnings were in the code from the beginning so that was nothing I
> introduced.
>
>
Thank you for the fix !

**Comment by miniupnp** [comment]
I rebased myself

**Pull request closed without merging**